### PR TITLE
[#272] SavedView thumbnail upload to S3

### PR DESCRIPTION
**Pull request by jswk**
- Remove saved_view.thumbnail column, as it inferred from other fields.
- Add properties, which configure S3 client and file storage.
- Validate thumbnail format in CreateSavedViewRequest.
- Create FileStorage, which uploads objects to S3 and deletes them.

Closes: #272.

### CODE REVIEW

File: s4e-backend/pom.xml [hunk 1]
```diff
@@ -26,6 +26,7 @@
         <greenmail.version>1.5.11</greenmail.version>
         <slugify.version>2.4</slugify.version>
         <springdoc-openapi.version>1.2.17</springdoc-openapi.version>
+        <awssdk.version>2.10.32</awssdk.version>
```
Comment (kmarszalek) [inline s4e-backend/pom.xml]: why this version?

File: s4e-backend/src/test/java/pl/cyfronet/s4e/controller/SavedViewControllerIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
...1wqWHMl5MkDeHelHVkf7xG0yMD9mj/AOFyNx3ZbAetMmC2JhJA0axNHimt1UoYlet8MWWx0fkG7jakbzlxrHDrqMpyZm8SuDdWJv2uGxtfQ3pjx2UHMjSqwbsm6tGw7mAJC280PPnaubcRmCOo/wC4+88+z8R1qM3WrmTUnMqnSRfLXlyqw/Qb06qpMUnWre5VHbPhvFwqe3AfowNVR+2l24mu51PxevzPfS7ZmLKMGVrMGuGHh9akHMcRkT023Q3hixsPWJqpurow7LeJWBHECuo7iGqlntgdHQwOMSWJETBTq7IigBYp1VevjUAcK6q69wz27qJug3pd+zMBIzdUzr1l2uummZfd+vOpL9rPBfbtlLLDG8hinErZLdbHhjHKJ2yHVwPuywGoGovaorDlTIblyuJXvezchHJZGVHN+HXq2Y/LsgtQFtDYc0JIETnuzAE5u7ht5+dFCb3SL2rONO12ud+0NfzvTlhd7YntmzRv8Wqa+6w/uKO3aFfHH9zL6T6vqUGLBuHv5/3xACHEGI6Kom75QoLRekWbtOb2zHQd1udJ8XtYNo+bTxc+XaLnm1uRY1JWO2fGyXyo65bhhYroG1uuv0odl2LHHyW/AwynVWzDxX0a/K1DLNMy+0LJ9XqccA59oFYhx3qrL71x+v8AWkb45VtZlHnrQ5OmpHizEEfFm5WrGLw7Llv4kDDUHhblf3b+RqFeOIXBMfRtlRoWuptyF8rDsnWwa3I+YYikO0Ntlicq5R3HxLftC/iHlfkNKaSK1p0U7LiGHip2wO1ALZm+elMdYrkUMBvOpY9oIL295u6319al7d72jJ1yRqqLEiAa3z5l0Tq7H3Lqb881/Sq5Uu2PPldT8Q/+/SorasqZwjMmPHYNDey2OVRcHxZFL6H4iBQ7jtnnMwVlNrejNm7OnKl+F3oR78POQtoeypy6WOumUD5U3RYzMxJtnLdnQckbL5eI1q7HBmNqqdCQREeGxrxm6u6fI2zf2YfOjHBYhpIldmu5vdvlp3UOtYA99kFvd4UYd/lc0+7sm+GT+Z/1ofqcbZNZjG7ECtvYFjK+RED5hx3s2W3O3K5a9yKY9rbMEa5eIy6H4VXy+tSauGBbX/20NCG9GHBZv5v8y0GfCtgTSaSzdUsCWH0rU0qxKsps3L/lpKRThLM1NYra1a06KfVkGsV9Sin/2Q==";
```
Comment (kmarszalek) [inline s4e-backend/src/test/java/pl/cyfronet/s4e/controller/SavedViewControllerIntegrationTest.java]: name

File: s4e-backend/src/test/java/pl/cyfronet/s4e/controller/SavedViewControllerTest.java [hunk 3]
```diff
@@ -34,7 +43,8 @@
...
...1wqWHMl5MkDeHelHVkf7xG0yMD9mj/AOFyNx3ZbAetMmC2JhJA0axNHimt1UoYlet8MWWx0fkG7jakbzlxrHDrqMpyZm8SuDdWJv2uGxtfQ3pjx2UHMjSqwbsm6tGw7mAJC280PPnaubcRmCOo/wC4+88+z8R1qM3WrmTUnMqnSRfLXlyqw/Qb06qpMUnWre5VHbPhvFwqe3AfowNVR+2l24mu51PxevzPfS7ZmLKMGVrMGuGHh9akHMcRkT023Q3hixsPWJqpurow7LeJWBHECuo7iGqlntgdHQwOMSWJETBTq7IigBYp1VevjUAcK6q69wz27qJug3pd+zMBIzdUzr1l2uummZfd+vOpL9rPBfbtlLLDG8hinErZLdbHhjHKJ2yHVwPuywGoGovaorDlTIblyuJXvezchHJZGVHN+HXq2Y/LsgtQFtDYc0JIETnuzAE5u7ht5+dFCb3SL2rONO12ud+0NfzvTlhd7YntmzRv8Wqa+6w/uKO3aFfHH9zL6T6vqUGLBuHv5/3xACHEGI6Kom75QoLRekWbtOb2zHQd1udJ8XtYNo+bTxc+XaLnm1uRY1JWO2fGyXyo65bhhYroG1uuv0odl2LHHyW/AwynVWzDxX0a/K1DLNMy+0LJ9XqccA59oFYhx3qrL71x+v8AWkb45VtZlHnrQ5OmpHizEEfFm5WrGLw7Llv4kDDUHhblf3b+RqFeOIXBMfRtlRoWuptyF8rDsnWwa3I+YYikO0Ntlicq5R3HxLftC/iHlfkNKaSK1p0U7LiGHip2wO1ALZm+elMdYrkUMBvOpY9oIL295u6319al7d72jJ1yRqqLEiAa3z5l0Tq7H3Lqb881/Sq5Uu2PPldT8Q/+/SorasqZwjMmPHYNDey2OVRcHxZFL6H4iBQ7jtnnMwVlNrejNm7OnKl+F3oR78POQtoeypy6WOumUD5U3RYzMxJtnLdnQckbL5eI1q7HBmNqqdCQREeGxrxm6u6fI2zf2YfOjHBYhpIldmu5vdvlp3UOtYA99kFvd4UYd/lc0+7sm+GT+Z/1ofqcbZNZjG7ECtvYFjK+RED5hx3s2W3O3K5a9yKY9rbMEa5eIy6H4VXy+tSauGBbX/20NCG9GHBZv5v8y0GfCtgTSaSzdUsCWH0rU0qxKsps3L/lpKRThLM1NYra1a06KfVkGsV9Sin/2Q==";
```
Comment (kmarszalek) [inline s4e-backend/src/test/java/pl/cyfronet/s4e/controller/SavedViewControllerTest.java]: name

File: s4e-backend/pom.xml (hunk 1 above)
Comment (jswk) [reply s4e-backend/pom.xml]: It was the latest when I added the dependency. They have a short release cycle

**Pull request merged**